packages/common/src/api/tan-query/useAudioTransactions.ts

```diff
@@ -0,0 +1,146 @@
+import { removeNullable } from '~/utils/typeUtils'
+
+import { QUERY_KEYS } from './queryKeys'
+import { Config } from './types'
+import { useCurrentUserId } from './useCurrentUserId'
+import { useUsers } from './useUsers'
+
+type GetAudioTransactionsArgs = {
+  offset?: number
+  limit?: number
+  sortMethod?: full.GetAudioTransactionsSortMethodEnum
+  sortDirection?: full.GetAudioTransactionsSortDirectionEnum
+}
+
+const parseTransaction = (tx: full.TransactionDetails): TransactionDetails => {
+  const transactionTypeMap: Record<string, TransactionType> = {
+    purchase_stripe: TransactionType.PURCHASE,
+    purchase_coinbase: TransactionType.PURCHASE,
+    purchase_unknown: TransactionType.PURCHASE,
+    'purchase unknown': TransactionType.PURCHASE,
+    tip: TransactionType.TIP,
+    user_reward: TransactionType.CHALLENGE_REWARD,
+    trending_reward: TransactionType.TRENDING_REWARD,
+    transfer: TransactionType.TRANSFER
+  }
+
+  const txType = transactionTypeMap[tx.transactionType]
+  switch (txType) {
+    case TransactionType.CHALLENGE_REWARD:
+    case TransactionType.TRENDING_REWARD:
+      return {
+        signature: tx.signature,
+        transactionType: txType,
+        method: TransactionMethod.RECEIVE,
+        date: tx.transactionDate,
+        change: tx.change,
+        balance: tx.balance,
+        metadata: tx.metadata as unknown as string
+      }
+    case TransactionType.PURCHASE:
+      return {
+        signature: tx.signature,
+        transactionType: txType,
+        method:
+          tx.method === 'purchase_stripe'
+            ? TransactionMethod.STRIPE
+            : tx.method === 'purchase_coinbase'
+              ? TransactionMethod.COINBASE
+              : TransactionMethod.RECEIVE,
+        date: tx.transactionDate,
+        change: tx.change,
+        balance: tx.balance,
+        metadata: undefined
+      }
+    case TransactionType.TIP:
+    case TransactionType.TRANSFER:
+      return {
+        signature: tx.signature,
+        transactionType: txType,
+        method:
+          tx.method === 'send'
+            ? TransactionMethod.SEND
+            : TransactionMethod.RECEIVE,
+        date: tx.transactionDate,
+        change: tx.change,
+        balance: tx.balance,
+        metadata: tx.metadata as unknown as string
+      }
+    default:
+      throw new Error('Unknown Transaction')
+  }
+}
+
+export const useAudioTransactions = (
+  args: GetAudioTransactionsArgs,
+  config?: Config
+) => {
+  const { audiusSdk } = useAudiusQueryContext()
+  const { data: userId } = useCurrentUserId()
+  const { offset = 0, limit = 10, sortMethod, sortDirection } = args
+
+  // Extract user IDs from transactions and fetch user data
+  const query = useQuery({
+    queryKey: [
+      QUERY_KEYS.audioTransactions,
+      userId,
+      offset,
+      limit,
+      sortMethod,
+      sortDirection
+    ],
+    queryFn: async () => {
+      if (!userId) return { txDetails: [], userIds: [] }
```

Comment:
in general, do we want the guard even when we have the enabled check?

I've been skipping it where it doesn't cause a type error, but could do either. Maybe worth deciding and norming on one or the other

Reply:
yeah im down to norm as well. happy to just userId! here though